core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCTemplate.scala

```diff
@@ -0,0 +1,364 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.ContractOraclePair._
+import org.bitcoins.core.protocol.dlc.DLCTemplate._
+import org.bitcoins.core.protocol.tlv._
+
+sealed trait DLCTemplate {
+  def totalCollateral: CurrencyUnit
+
+  def oracles: Vector[OracleAnnouncementTLV]
```

Comment:
So my thoughts on this are a bit different. 

It seems that a template shouldn't have any knowledge of oracles. It seems that a logical process for building a DLC are 

1. Specifying the contract info
2. Pick the oracles for the contract